FeaturedArticleWidget/FeaturedArticleWidget.swift

```diff
@@ -166,17 +183,24 @@ class FeaturedArticleWidget: ExtensionViewController, NCWidgetProviding {
     }
     
     @objc func saveButtonPressed() {
-        guard let article = self.article, let articleKey = article.key else {
+        guard let articleKey = articleURL?.wmf_databaseKey else {
             return
         }
-        let isSaved = dataStore?.savedPageList.toggleSavedPage(forKey: articleKey) ?? false
-        expandedArticleView.saveButton.saveButtonState = isSaved ? .longSaved : .longSave
+        WidgetController.shared.startWidgetUpdateTask { (done: Bool) in
+            
```

Comment:
Might be good to have a `DDLogDebug` here similar to how you did `DDLogDebug("Widget did finish")` for the empty `userCompletion` block below.